central/sensor/service/connection/connection_impl.go

```diff
@@ -155,8 +156,13 @@ func (c *sensorConnection) runRecv(ctx context.Context, grpcServer central.Senso
 
 func (c *sensorConnection) handleMessages(ctx context.Context, queue *dedupingQueue) {
 	for msg := queue.pullBlocking(&c.stopSig); msg != nil; msg = queue.pullBlocking(&c.stopSig) {
-		if err := c.handleMessage(ctx, msg); err != nil {
-			log.Errorf("Error handling sensor message: %v", err)
+		err := safe.Run(func() {
+			if err := c.handleMessage(ctx, msg); err != nil {
+				log.Errorf("Error handling sensor message: %v", err)
+			}
+		})
+		if err != nil {
+			log.Errorf("UNEXPECTED panic in handle message: %v", err)
```

Comment:
Run already logs this (well technically `utils.ShouldErr`) and will panic on dev

Reply:
true, I removed the UNEXPECTED portion